modules/custom-css/migrate-to-core.php

```diff
@@ -0,0 +1,241 @@
+			if ( empty( $stylesheet ) ) {
+				continue;
+			}
+
+			$migrated[] = $post->ID;
+			$preprocessor = get_post_meta( $post->ID, 'custom_css_preprocessor', true );
+			$css = $post->post_content;
+			$pre = null;
+
+			if ( $preprocessor && isset( $preprocessors[ $preprocessor ] ) ) {
+				$preprocessor = 'less';
+				$pre = $css;
+				$css = call_user_func( $preprocessors[ $preprocessor ]['callback'], $pre );
+			}
+
+			// Do we need to remove any filters here for users without `unfiltered_html` ?
+
+			// Format here into calls to wp_insert_post() or wp_update_post()
+			$args = array(
+				'post_content'          => $css,
+				'post_content_filtered' => $pre,
+				'post_title'            => $stylesheet,
+				'post_name'             => sanitize_title( $stylesheet ),
+				'post_type'             => 'custom_css',
+				'post_status'           => 'publish',
+				'post_author'           => $post->post_author,
+				'post_modified'         => $post->post_modified,
+				'post_modified_gmt'     => $post->post_modified_gmt,
+			);
+
+			// Run one quick query to determine if we have a post for that theme.
+			if ( empty( $themes_posts[ $stylesheet ] ) ) {
+				$core_post = wp_get_custom_css_post( $stylesheet );
+				$themes_posts[ $stylesheet ] = ( $core_post instanceof WP_Post ) ? $core_post->ID : null;
+			}
+
+			if ( $themes_posts[ $stylesheet ] ) {
+				$args['ID'] = $themes_posts[ $stylesheet ];
+				// We already know the ID, so no need to save the return value.
+				wp_update_post( wp_slash( $args ) );
+			} else {
+				// This is a new post we're stashing it under, so save the ID.
+				$themes_posts[ $stylesheet ] = wp_insert_post( wp_slash( $args ) );
+			}
+		}
+
+		// Migrate the settings from revision meta to theme mod.
+		$options = self::get_options( $jetpack_css_revision->ID );
+		set_theme_mod( 'jetpack_custom_css', $options );
+
+		// If we've migrated some CSS for the current theme and there was already something there in the Core dataset ...
+		if ( isset( $jetpack_latest_theme_revisions[ $current_stylesheet ] ) && $core_css_post && $jetpack_css_revision ) {
+			$preprocessor = $options['preprocessor'];
+
+			$css = $core_css_post->post_content;
+			$pre = $core_css_post->post_content_filtered;
+			if ( $preprocessor ) {
+				if ( $pre ) {
+					$pre .= "\r\n\r\n/*\r\n\t" . esc_js( __( 'CSS Migrated from Jetpack:', 'jetpack' ) ) . "\r\n*/\r\n\r\n";
+				}
+				$pre .= $jetpack_css_revision->post_content_filtered;
+
+				$css .= "\r\n\r\n/*\r\n\t" . esc_js( __( 'CSS Migrated from Jetpack:', 'jetpack' ) ) . "\r\n*/\r\n\r\n";
+				$css .= call_user_func( $preprocessors[ $preprocessor ]['callback'], $jetpack_css_revision->post_content_filtered );
+			} else {
+				$css .= "\r\n\r\n/*\r\n\t" . esc_js( __( 'CSS Migrated from Jetpack:', 'jetpack' ) ) . "\r\n*/\r\n\r\n";
+				$css .= $jetpack_css_revision->post_content;
+			}
+
+			wp_update_post( array(
```

Comment:
The `array` needs to be wrapped in `wp_slash()` since `wp_update_post()` expects (aggravatingly) its input to be pre-slashed. So if `wp_slash()` isn't added, then any slashes in `$css` or `$pre` will get dropped and a CSS parse error could result.